bcda/models/models.go

```diff
@@ -29,6 +28,24 @@ const BCDA_FHIR_MAX_RECORDS_EOB_DEFAULT = 200
 const BCDA_FHIR_MAX_RECORDS_PATIENT_DEFAULT = 5000
 const BCDA_FHIR_MAX_RECORDS_COVERAGE_DEFAULT = 4000
 
+// NOTE: This should be temporary, we should get to the point where this file only contains data models. Once that happens,
+// we no longer have the need for the data models tor produce other data models and we can remove reference to the service.
+var (
+	serviceInstance Service // Singleton service instance
+	once            sync.Once
+)
+
+// GetService returns the singleton instance of Service. It creates the service if it has not been created before.
```

Comment:
Curious how this works in a multi-threaded env, like we have in AWS.

Reply:
sync.Once wraps the exclusive lock (and I think an atomic bool?) to ensure it only runs once. So we should be safe here.